### PR TITLE
[Improvement-5880][api] Optimized data structure of pagination query API results

### PR DESCRIPTION
**Pull request by soreak**
[5880][refactor]Optimized data structure of pagination query API results
- refactor PageInfo and delete returnDataListPaging in API
- modify the related Controller and Service and the corresponding Test 
- this closes [Improvement][API] Page paging query API result data structure optimization #5880

### CODE REVIEW

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/SchedulerServiceImpl.java [hunk 1]
```diff
@@ -415,15 +416,18 @@
...
 
         Project project = projectMapper.queryByName(projectName);
 
+        Map<String, Object> checkResult = new HashMap<>();
         // check project auth
-        boolean hasProjectAndPerm = projectService.hasProjectAndPerm(loginUser, project, result);
+        boolean hasProjectAndPerm = projectService.hasProjectAndPerm(loginUser, project, checkResult);
         if (!hasProjectAndPerm) {
+            Status status = (Status) checkResult.get(Constants.STATUS);
+            putMsg(result,status);
```
Comment (zhuangchong) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/SchedulerServiceImpl.java]: The exception information is not added to the Result.

**Review by zhuangchong — CHANGES_REQUESTED** [review]
Please check other similar situations.

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/SchedulerServiceImpl.java (hunk 1 above)
Comment (soreak) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/SchedulerServiceImpl.java]: ok,i'll fix it

**Review by CalvinKirs — COMMENTED** [review]
Just my personal opinion: servece should only contain specific data, Result is generally the encapsulated carrier of external interaction data, should not be reflected in the service. Related errors can be directly thrown exceptions, API unified exception capture interception. This is a historical legacy, but it would be better if it could be solved at once.

**Comment by zhuangchong** [comment]
> Just my personal opinion: servece should only contain specific data, Result is generally the encapsulated carrier of external interaction data, should not be reflected in the service. Related errors can be directly thrown exceptions, API unified exception capture interception. This is a historical legacy, but it would be better if it could be solved at once.

@CalvinKirs @soreak 

I also agree with this point, this change involves a lot of code, it is recommended to start a new PR for optimization.

**Review by zhuangchong — APPROVED** [review]
LGTM.

**Comment by brave-lee** [comment]
I think we should discuss the rationality of such a change in the API

**Comment by zhuangchong** [comment]
> I think we should discuss the rationality of such a change in the API

The current situation of the paging interface is that there are two data structures, and the current change is to change to a common data structure.

**Comment by brave-lee** [comment]
Common methods should be modified here, not every api internal method.

**Comment by zhuangchong** [comment]
> Common methods should be modified here, not every api internal method.

There are currently several situations, resulting in the need to modify each Service layer
1. Multiple controllers use Page paging query
2. The original logic is to obtain PageInfo data at the controller layer and convert it into a new data structure. The PageInfo data structure is reconstructed here, and all the Service layers involved need to be modified.
3. A small part of the controller layer directly returns the PageInfo data structure.

**Comment by zhuangchong** [comment]
this issue: #5880

**Comment by brave-lee** [comment]
The Result structure should not be returned in the service layer

**Comment by brave-lee** [comment]
BTW,the code for API modifications should be pushed to the json_split_two branch

**Comment by zhuangchong** [comment]
> The Result structure should not be returned in the service layer

Yes, considering that this change involves too much code, and this contributor contributed for the first time, let him submit a new PR to optimize this problem.

**Comment by brave-lee** [comment]
For PR with improved api module, please submit to the json_split_two branch, this branch leads dev a lot.

**Comment by zhuangchong** [comment]
> For PR with improved api module, please submit to the json_split_two branch, this branch leads dev a lot.

ok,

**Pull request merged**